hugolib/page.go

```diff
@@ -1114,11 +1114,11 @@ func (p *Page) prepareForRender(cfg *BuildCfg) error {
 
 var ErrHasDraftAndPublished = errors.New("both draft and published parameters were found in page's frontmatter")
 
-func (p *Page) update(f interface{}) error {
-	if f == nil {
-		return errors.New("no metadata found")
+func (p *Page) update(frontmatter interface{}) error {
```

Comment:
why is frontmatter an `interface{}` and not `map[string]interface{}`?

Reply:
That's what it was before / I didn't change it. I'm a novice go programmer, so I don't change things I don't need to yet :)

Reply:
Want me to change it now?

Reply:
OK, no just leave it.

Reply:
But you need to do something about the commit messages. 5 commits is maybe too much, too, but the important part is the commit message title (we reference the issues in the body with "Fixes #<some-id>, See #<some-id>.

Reply:
I'd recommend updating the function signature to use `map[string]interface{}` instead of `interface{}`.  We type-assert as `map[string]interface{}` at L1121 and don't handle any assertion failures there.

Reply:
>I'd recommend updating the function signature to use map[string]interface{} instead of interface{}

We will consider that for the future, not this PR.

Reply:
How about I squash it down to 2 commits (add test coverage + fix)? And since I'm doing that anyway I'll go ahead and change the method signature.

Reply:
So ditch Pull Request #4321 (adds pure test coverage)? If we keep it, then I can get the total (counting #4321) down to 3 commits. I don't care either way.

Reply:
No, keep the code, tests included. This is not a "squash everything" game;  just get the commit messages in line with the guidelines (re my comment above).